pulsar-client-cpp/lib/PartitionedProducerImpl.cc

```diff
@@ -87,13 +87,18 @@ unsigned int PartitionedProducerImpl::getNumPartitionsWithLock() const {
     return getNumPartitions();
 }
 
-ProducerImplPtr PartitionedProducerImpl::newInternalProducer(unsigned int partition) const {
+ProducerImplPtr PartitionedProducerImpl::newInternalProducer(unsigned int partition) {
     using namespace std::placeholders;
     std::string topicPartitionName = topicName_->getTopicPartitionName(partition);
     auto producer = std::make_shared<ProducerImpl>(client_, topicPartitionName, conf_, partition);
-    producer->getProducerCreatedFuture().addListener(
-        std::bind(&PartitionedProducerImpl::handleSinglePartitionProducerCreated,
-                  const_cast<PartitionedProducerImpl*>(this)->shared_from_this(), _1, _2, partition));
+
+    if (conf_.getLazyStartPartitionedProducers()) {
```

Comment:
This looks like a noop. Rather than having "createLazyPartitionProducer" count the number of producers created and change state when all done, you could change ::start() to change the state.

Reply:
It's not a no-op as it kicks off the partition update task which is something we want running, even if no producers have started. It seems the most simple way of doing it.

Reply:
What I was suggesting was to kick off the update task in ::start() if lazyproducer is enabled. Is numProducersCreated used for anything else?

Reply:
Yes, it's required when closing. Also, this method will be called when the partition update task runs and see a new number of partitions, so it fits well with that.

Reply:
:+1: